### ISSUE TITLE
Dependencies conflictL perl v22 and perl v18 for snoscan

### ISSUE THREAD

**Issue by aksenia** [conda/conda#7358]
**I'm submitting a...**
  - [ ] bug report

### Steps to Reproduce
```
conda install snoscan=0.9b
Solving environment: failed

UnsatisfiableError: The following specifications were found to be in conflict:
  - perl-local-lib -> perl-extutils-install -> perl=5.18.2
  - snoscan=0.9b

```

## Environment Information
```
active environment : None
       user config file : /home/kselav/.condarc
 populated config files : /home/kselav/.condarc
          conda version : 4.5.4
    conda-build version : 3.8.0
         python version : 2.7.13.final.0
       base environment : /home/kselav/anaconda2  (writable)
           channel URLs : https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://conda.binstar.org/faircloth-lab/linux-64
                          https://conda.binstar.org/faircloth-lab/noarch
                          https://conda.anaconda.org/r/linux-64
                          https://conda.anaconda.org/r/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/kselav/anaconda2/pkgs
                          /home/kselav/.conda/pkgs
       envs directories : /home/kselav/anaconda2/envs
                          /home/kselav/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.4 requests/2.7.0 CPython/2.7.13 Linux/3.16.6-2-desktop opensuse/13.2 glibc/2.19
                UID:GID : 3122:676
             netrc file : None
           offline mode : False
```

</p></details>

```
==> /home/kselav/.condarc <==
ssl_verify: False
channels:
  - bioconda
  - https://conda.binstar.org/faircloth-lab
  - r
  - defaults
  - conda-forge
```

```
_license                  1.1                      py27_1    defaults
alabaster                 0.7.10                   py27_0    defaults
anaconda                  custom           py27h4a00acb_0    defaults
anaconda-client           1.6.3                    py27_0    defaults
anaconda-navigator        1.6.2                    py27_0    defaults
anaconda-project          0.6.0                    py27_0    defaults
argcomplete               1.0.0                    py27_1    defaults
asn1crypto                0.22.0                   py27_0    defaults
astroid                   1.4.9                    py27_0    defaults
astropy                   1.3.2               np112py27_0    defaults
babel                     2.4.0                    py27_0    defaults
backports                 1.0                      py27_0    defaults
backports.functools_lru_cache 1.5                      py27_1    defaults
backports_abc             0.5                      py27_0    defaults
beautifulsoup4            4.4.1                     <pip>
beautifulsoup4            4.6.0                    py27_0    defaults
bitarray                  0.8.1                    py27_0    defaults
blast                     2.7.1               boost1.64_1    bioconda
blaze                     0.10.1                   py27_0    defaults
bleach                    1.5.0                    py27_0    defaults
bokeh                     0.12.5                   py27_1    defaults
boost                     1.64.0                   py27_4    conda-forge
boost-cpp                 1.64.0                        1    conda-forge
boto                      2.46.1                   py27_0    defaults
bottleneck                1.2.1               np112py27_0    defaults
bowtie                    1.2.0                    py27_0    bioconda
bowtie2                   2.3.0                    py27_1    bioconda
bzip2                     1.0.6                         1    conda-forge
ca-certificates           2018.03.07                    0    defaults
cairo                     1.14.6                        4    conda-forge
cdecimal                  2.3                      py27_2    defaults
certifi                   0.0.8                    py27_0    dan_blanchard
cffi                      1.10.0                   py27_0    defaults
chardet                   3.0.3                    py27_0    defaults
chest                     0.2.3                    py27_0    defaults
click                     6.7                      py27_0    defaults
cloudpickle               0.2.2                    py27_0    defaults
clyent                    1.2.2                    py27_0    defaults
collectl                  4.0.4                         1    bioconda
colorama                  0.3.9                    py27_0    defaults
conda                     4.5.4                    py27_0    defaults
conda-build               3.8.0                    py27_0    defaults
conda-env                 2.6.0                h36134e3_1    defaults
conda-verify              2.0.0                    py27_0    defaults
configobj                 5.0.6                    py27_0    defaults
configparser              3.5.0                    py27_0    defaults
contextlib2               0.5.5                    py27_0    defaults
cryptography              1.8.1                    py27_0    defaults
curl                      7.52.1                        0    defaults
cycler                    0.10.0                   py27_0    defaults
cython                    0.25.2                   py27_0    defaults
cytoolz                   0.8.2                    py27_0    defaults
dask                      0.14.3                   py27_1    defaults
datashape                 0.5.4                    py27_0    defaults
dbus                      1.10.10                       0    defaults
decorator                 4.0.11                   py27_0    defaults
dill                      0.2.5                    py27_0    defaults
distributed               1.16.3                   py27_0    defaults
docutils                  0.13.1                   py27_0    defaults
dxpy                      0.254.0                   <pip>
entrypoints               0.2.2                    py27_1    defaults
enum34                    1.1.6                    py27_0    defaults
et_xmlfile                1.0.1                    py27_0    defaults
expat                     2.1.0                         0    defaults
fastcache                 1.0.2                    py27_1    defaults
fastool                   0.1.4                         1    bioconda
filelock                  2.0.7                    py27_0    defaults
flask                     0.12.2                   py27_0    defaults
flask-cors                3.0.2                    py27_0    defaults
fontconfig                2.12.1                        5    conda-forge
freetype                  2.7                           1    conda-forge
funcsigs                  1.0.2                    py27_0    defaults
functools32               3.2.3.2                  py27_0    defaults
fusepy                    2.0.2                     <pip>
futures                   3.0.4                     <pip>
futures                   3.1.1                    py27_0    defaults
gatk                      3.7                      py27_0    bioconda
get_terminal_size         1.0.0                    py27_0    defaults
gettext                   0.19.8.1             hd7bead4_3    defaults
gevent                    1.2.1                    py27_0    defaults
glib                      2.51.4                        0    conda-forge
glob2                     0.4.1                    py27_0    bioconda
globus-sdk                1.1.1                     <pip>
gmp                       6.1.2                h6c8ec71_1    defaults
gnutls                    3.5.15                        0    conda-forge
greenlet                  0.4.12                   py27_0    defaults
grin                      1.2.1                    py27_3    defaults
gst-plugins-base          1.8.0                         0    defaults
gstreamer                 1.8.0                         0    defaults
h5py                      2.7.0               np112py27_0    defaults
harfbuzz                  1.3.4                         2    conda-forge
hdf5                      1.8.17                        1    defaults
heapdict                  1.0.0                    py27_1    defaults
html5lib                  0.999                    py27_0    defaults
icu                       58.2                 h9c2bf20_1    defaults
idna                      2.5                      py27_0    defaults
imagesize                 0.7.1                    py27_0    defaults
ipaddress                 1.0.18                   py27_0    defaults
ipykernel                 4.6.1                    py27_0    defaults
ipython                   5.3.0                    py27_0    defaults
ipython_genutils          0.2.0                    py27_0    defaults
ipywidgets                6.0.0                    py27_0    defaults
isort                     4.2.5                    py27_0    defaults
itsdangerous              0.24                     py27_0    defaults
java-jdk                  8.0.92                        1    bioconda
jbig                      2.1                           0    defaults
jdcal                     1.3                      py27_0    defaults
jedi                      0.10.2                   py27_2    defaults
jellyfish                 2.2.6                         0    bioconda
jinja2                    2.9.6                    py27_0    defaults
joblib                    0.11                      <pip>
jpeg                      9b                            0    defaults
jsonschema                2.6.0                    py27_0    defaults
jupyter                   1.0.0                    py27_3    defaults
jupyter_client            5.0.1                    py27_0    defaults
jupyter_console           5.1.0                    py27_0    defaults
jupyter_core              4.3.0                    py27_0    defaults
lazy-object-proxy         1.2.2                    py27_0    defaults
libffi                    3.2.1                         1    defaults
libgcc                    4.8.5                         2    defaults
libgcc-ng                 7.2.0                hdf63c60_3    defaults
libgfortran               3.0.0                         1    defaults
libiconv                  1.14                          0    defaults
libidn11                  1.33                          0    conda-forge
libpng                    1.6.30                        1    defaults
libsodium                 1.0.10                        0    defaults
libstdcxx-ng              7.2.0                hdf63c60_3    defaults
libtiff                   4.0.6                         3    defaults
libtool                   2.4.2                         0    defaults
libxcb                    1.12                          1    defaults
libxml2                   2.9.4                         0    defaults
libxslt                   1.1.29                        0    defaults
llvmlite                  0.18.0                   py27_0    defaults
locket                    0.2.0                    py27_1    defaults
lxml                      3.7.3                    py27_0    defaults
mafft                     7.313                         0    bioconda
markupsafe                0.23                     py27_2    defaults
matplotlib                2.1.1                    py27_0    conda-forge
mistune                   0.7.4                    py27_0    defaults
mkl                       2017.0.1                      0    defaults
mkl-service               1.1.2                    py27_3    defaults
mpmath                    0.19                     py27_1    defaults
msgpack-python            0.4.8                    py27_0    defaults
multipledispatch          0.4.9                    py27_0    defaults
navigator-updater         0.1.0                    py27_0    defaults
nbconvert                 5.1.1                    py27_0    defaults
nbformat                  4.3.0                    py27_0    defaults
nettle                    3.3                           0    conda-forge
networkx                  1.11                     py27_0    defaults
nltk                      3.2.3                    py27_0    defaults
nose                      1.3.7                    py27_1    defaults
notebook                  5.0.0                    py27_0    defaults
numba                     0.33.0              np112py27_0    defaults
numexpr                   2.6.2               np112py27_0    defaults
numpy                     1.12.1                   py27_0    defaults
numpydoc                  0.6.0                    py27_0    defaults
odo                       0.5.0                    py27_1    defaults
olefile                   0.44                     py27_0    defaults
openjdk                   8.0.121                       1    defaults
openpyxl                  2.4.7                    py27_0    defaults
openssl                   1.0.2o               h20670df_0    defaults
packaging                 16.8                     py27_0    defaults
pandas                    0.20.1              np112py27_0    defaults
pandocfilters             1.4.1                    py27_0    defaults
pango                     1.40.3                        2    conda-forge
parafly                   r2013_01_21                   0    bioconda
partd                     0.3.8                    py27_0    defaults
patchelf                  0.9                           0    defaults
path.py                   10.3.1                   py27_0    defaults
pathlib2                  2.2.1                    py27_0    defaults
patsy                     0.4.1                    py27_0    defaults
pcre                      8.39                          1    defaults
pep8                      1.7.0                    py27_0    defaults
perl                      5.18.2                        8    dan_blanchard
perl-app-cpanminus        1.7001               pl5.18.2_1    dan_blanchard
perl-archive-tar          1.90                 pl5.18.2_1    dan_blanchard
perl-carp                 1.29                 pl5.18.2_1    dan_blanchard
perl-compress-raw-bzip2   2.069                         1    bioconda
perl-compress-raw-zlib    2.069                         3    bioconda
perl-cpan                 2.00                 pl5.18.2_1    dan_blanchard
perl-cpan-meta            2.132830             pl5.18.2_1    dan_blanchard
perl-cpan-meta-requirements 2.122                pl5.18.2_1    dan_blanchard
perl-cpan-meta-yaml       0.011                pl5.18.2_0    dan_blanchard
perl-data-dumper          2.145                pl5.18.2_1    dan_blanchard
perl-encode               2.49                 pl5.18.2_1    dan_blanchard
perl-exporter             5.68                 pl5.18.2_1    dan_blanchard
perl-exporter-tiny        0.042                         1    bioconda
perl-extutils-cbuilder    0.280210             pl5.18.2_1    dan_blanchard
perl-extutils-install     1.59                 pl5.18.2_1    dan_blanchard
perl-extutils-makemaker   6.66                 pl5.18.2_1    dan_blanchard
perl-extutils-manifest    1.63                 pl5.18.2_1    dan_blanchard
perl-extutils-parsexs     3.18                 pl5.18.2_1    dan_blanchard
perl-file-path            2.09                 pl5.18.2_1    dan_blanchard
perl-file-temp            0.23                 pl5.18.2_1    dan_blanchard
perl-getopt-long          2.39                 pl5.18.2_1    dan_blanchard
perl-io                   1.28                 pl5.18.2_1    dan_blanchard
perl-io-compress          2.064                pl5.18.2_1    dan_blanchard
perl-io-zlib              1.10                          1    bioconda
perl-json-pp              2.27202              pl5.18.2_1    dan_blanchard
perl-list-moreutils       0.33                 pl5.18.2_1    dan_blanchard
perl-local-lib            2.000007             pl5.18.2_1    dan_blanchard
perl-module-build         0.4205               pl5.18.2_0    dan_blanchard
perl-module-metadata      1.000019             pl5.18.2_0    dan_blanchard
perl-parent               0.225                pl5.18.2_1    dan_blanchard
perl-parse-cpan-meta      1.4413               pl5.18.2_0    dan_blanchard
perl-pathtools            3.40                 pl5.18.2_1    dan_blanchard
perl-perl-ostype          1.003                pl5.18.2_1    dan_blanchard
perl-pod-simple           3.28                 pl5.18.2_1    dan_blanchard
perl-podlators            v2.5.3               pl5.18.2_1    dan_blanchard
perl-scalar-list-utils    1.45                          2    bioconda
perl-test                 1.26                 pl5.18.2_1    dan_blanchard
perl-test-harness         3.30                 pl5.18.2_1    dan_blanchard
perl-test-simple          1.001002             pl5.18.2_1    dan_blanchard
perl-text-abbrev          1.02                 pl5.18.2_1    dan_blanchard
perl-text-parsewords      3.28                 pl5.18.2_1    dan_blanchard
perl-threaded             5.22.0                       10    bioconda
perl-version              0.9907               pl5.18.2_0    dan_blanchard
pexpect                   4.2.1                    py27_0    defaults
pickleshare               0.7.4                    py27_0    defaults
pillow                    4.3.0                    py27_0    conda-forge
pip                       9.0.1                    py27_1    defaults
pip                       10.0.1                    <pip>
pixman                    0.34.0                        0    defaults
pkginfo                   1.4.1                    py27_0    defaults
ply                       3.10                     py27_0    defaults
postgresql                9.5.4                         0    defaults
prompt_toolkit            1.0.14                   py27_0    defaults
psutil                    5.2.2                    py27_0    defaults
psutil                    3.3.0                     <pip>
ptyprocess                0.5.1                    py27_0    defaults
py                        1.4.33                   py27_0    defaults
pyasn1                    0.1.9                    py27_0    defaults
pycairo                   1.10.0                   py27_0    defaults
pycosat                   0.6.3            py27ha4109ae_0    defaults
pycparser                 2.17                     py27_0    defaults
pycrypto                  2.6.1                    py27_6    defaults
pycurl                    7.43.0                   py27_2    defaults
pyflakes                  1.5.0                    py27_0    defaults
pygments                  2.2.0                    py27_0    defaults
pylint                    1.6.4                    py27_1    defaults
pyodbc                    4.0.16                   py27_0    defaults
pyopenssl                 17.0.0                   py27_0    defaults
pyparsing                 2.1.4                    py27_0    defaults
pyqt                      5.6.0                    py27_2    defaults
pytables                  3.3.0               np112py27_0    defaults
pytest                    3.0.7                    py27_0    defaults
python                    2.7.13                        0    defaults
python-dateutil           2.6.0                    py27_0    defaults
python-dateutil           2.3                       <pip>
python-magic              0.4.6                     <pip>
pytz                      2017.2                   py27_0    defaults
pywavelets                0.5.2               np112py27_0    defaults
pyyaml                    3.12                     py27_0    defaults
pyzmq                     16.0.2                   py27_0    defaults
qt                        5.6.2                         3    conda-forge
qtawesome                 0.4.4                    py27_0    defaults
qtconsole                 4.3.0                    py27_0    defaults
qtpy                      1.2.1                    py27_0    defaults
quast                     4.5                       <pip>
readline                  6.2                           2    defaults
redis                     3.2.0                         0    defaults
redis-py                  2.10.5                   py27_0    defaults
requests                  2.7.0                     <pip>
requests                  2.14.2                   py27_0    defaults
rope                      0.9.4                    py27_1    defaults
ruamel_yaml               0.11.14                  py27_1    defaults
samtools                  1.4.1                         0    bioconda
scandir                   1.5                      py27_0    defaults
scikit-image              0.13.0              np112py27_0    defaults
scikit-learn              0.18.1              np112py27_1    defaults
scipy                     0.19.0              np112py27_0    defaults
seaborn                   0.7.1                    py27_0    defaults
setuptools                36.5.0           py27h68b189e_0    defaults
simplegeneric             0.8.1                    py27_1    defaults
simplejson                3.10.0                    <pip>
singledispatch            3.4.0.3                  py27_0    defaults
sip                       4.18                     py27_0    defaults
six                       1.10.0                   py27_0    defaults
slclust                   02022010                      1    bioconda
snowballstemmer           1.2.1                    py27_0    defaults
sockjs-tornado            1.0.3                    py27_0    defaults
sortedcollections         0.5.3                    py27_0    defaults
sortedcontainers          1.5.7                    py27_0    defaults
sphinx                    1.5.6                    py27_0    defaults
spyder                    3.1.4                    py27_0    defaults
sqlalchemy                1.1.9                    py27_0    defaults
sqlite                    3.13.0                        0    defaults
ssl_match_hostname        3.4.0.2                  py27_1    defaults
statsmodels               0.8.0               np112py27_0    defaults
subprocess32              3.2.7                    py27_0    defaults
sympy                     1.0                      py27_0    defaults
tbb                       2017_20170412                 0    conda-forge
tblib                     1.3.2                    py27_0    defaults
terminado                 0.6                      py27_0    defaults
testpath                  0.3                      py27_0    defaults
tk                        8.5.18                        0    defaults
toolz                     0.8.2                    py27_0    defaults
tornado                   4.5.1                    py27_0    defaults
traitlets                 4.3.2                    py27_0    defaults
trimmomatic               0.36                          3    bioconda
trinity                   2.4.0                         5    bioconda
unicodecsv                0.14.1                   py27_0    defaults
unixodbc                  2.3.4                         0    defaults
wcwidth                   0.1.7                    py27_0    defaults
werkzeug                  0.12.2                   py27_0    defaults
wheel                     0.29.0                   py27_0    defaults
widgetsnbextension        2.0.0                    py27_0    defaults
wrapt                     1.10.10                  py27_0    defaults
ws4py                     0.3.2                     <pip>
xattr                     0.6.4                     <pip>
xlrd                      1.0.0                    py27_0    defaults
xlsxwriter                0.9.6                    py27_0    defaults
xlwt                      1.2.0                    py27_0    defaults
xz                        5.2.2                         1    defaults
yaml                      0.1.6                         0    defaults
zeromq                    4.1.5                         0    defaults
zict                      0.1.2                    py27_0    defaults
zlib                      1.2.8                         3    defaults

```

**Comment by aksenia** [comment]
I can not wrap my head around this mysterious perl 18.2, Why does it appear in conda list? Where is my perl 22.0? Could this be a reason for the problem?

**Comment by mbargull** [comment]
> I can not wrap my head around this mysterious perl 18.2, \[...\] Could this be a  reason for the problem?

Yes, that's indeed the problem. The `conda list` output (thanks for providing it!) shows that at one point you must've installed a bunch of Perl packages from the `dan_blanchard` channel, including `perl 5.18.2=8` which the others depend on.

Your `conda config` output shows a few setting that I'd advice you to change:
* `ssl_verify: False` shouldn't usually be set all the time, rather just for troubleshooting.
  You can remove it by running
  `conda config --remove-key ssl_verify`
* You have `r` in the channel list, but the packages from there are included in `defaults` with `https://repo.anaconda.com/pkgs/r` (as you can see from your `conda info` output).
  You can remove it with
  `conda config --remove channels r`.
* You have `https://conda.binstar.org/faircloth-lab` in your channel list. Looks like that channel mostly includes (older versions of) packages also available in `bioconda`/`conda-forge`.
  If you don't need it, you can remove it with
  `conda config --remove channels 'https://conda.binstar.org/faircloth-lab'`
* The channel priorities of your configuration are `[bioconda, defaults, conda-forge]`. The official recommendation for Bioconda is `[bioconda, conda-forge, defaults]`.
  You can change that by running
  `conda config --add channels conda-forge --add channels bioconda`

More information regarding that last point:
About a year ago the [Bioconda documentation on setting up channel order](http://bioconda.github.io/#set-up-channels) were updated to recommend giving `conda-forge` a higher priority than `defaults`. This is because `bioconda` builds on top of `conda-forge`'s packages now.
But note that it is generally not advised, whenever possible, to mix and match packages from `conda-forge` and `defaults`, due to binary incompatibilities (though, the situation for this will likely improve in near future). This also applies (partially due to other reasons) to packages installed by `pip` (seeing that you also have some packages marked as installed by `<pip>` in your list) -- if the packages you need are available as Conda packages, you would want to prefer those over `pip install`ing those form PyPI.

After you updated your channel configuration with those `conda config` commands, you can try to run `conda update --all` which **might** cause those `dan_blanchard` packages to be replaced by `conda-forge` ones. If that doesn't help (which is not unlikely), one of the easiest solutions is just to start over with a fresh new Miniconda3 (or Anaconda3) installation. Regardless whether you start over again or not, I strongly advice you to read up on https://conda.io/docs/user-guide/tasks/manage-environments.html . Essentially, this allows you to create separate environment for different tools or workflows with which you can avoid version conflicts or those binary incompatibilities between `defaults` and `conda-forge`. You can even have per-environment configurations by using `conda config --env your-env-name`.

**Comment by aksenia** [comment]
Hi! Your advice was extremely helpful - thank you. The recipe you suggested helped with some but not all packages from `dan_blanchard` channel. After poking around I concluded that this channel must be old and does not have proper "noarch" folder and other required specification and is not recognized by config as a valid channel. 
`conda uninstall perl ` was able to remove perl and all associated packages from this older channel. I then reinstalled perl from `conda-forge`.
